### ISSUE TITLE
Add line numbers to syntax errors

### ISSUE THREAD

**Issue by jeremybradbury** [apidoc/apidoc#1088]
# Feature request

The problem: your syntax error points to an @apiUse line or other common occurrence in the file, so the syntax error is ambiguous requiring some digging for larger files. Right now syntax error reporting is about as specific as only sharing the name of the function in which it occurred, instead of the line. 

The solution: also share the line number in the file which the syntax error occurs. 

My thoughts: I very much appreciate how clean they are, but you can go too far sometimes. If this is (somehow) performance related, it's a common architecture pattern to "ignore performance when failing command line tools". We'd all rather wait for better errors, when things go wrong, than getting worse errors quickly. Sometimes that sentiment is taken further to slow down the success path slightly.

**Comment by NicolasCARPi** [comment]
I completely agree. The core part of the lib (previously apidoc/apidoc-core) hasn't been touched much during my rewrite and as far as I can tell there is no easy access to the line number.

**Comment by jeremybradbury** [comment]
Glad my DX (developer experience) complaints aren't useless. FWIW the fact I'm here complaining instead of using something else, says a lot about how I value the product. 

What come across as complaints, are actually compliments... like "but you only got these easily fixable gripes with fixes I could probably still live without".

Perhaps I can find some time to contribute PRs instead of just issues.
